### PR TITLE
DS-137 Add icon support in Menu component and a new display prop for Share component

### PR DESCRIPTION
**Pull request by mikemai2awesome**
## Jira

https://pegadigitalit.atlassian.net/browse/DS-137

## Summary

Adds icon support in Menu component and a new `display` prop for Share component.

## Details

1. `icon_before` and `icon_after` added to menu-item.
2. `display` prop added to Share component. `inline`and `menu` are available.
3. Revamped docs for Share component.
4. Revamped docs for Menu component.
5. Fixed display and alignment issues with Icon component.
6. Fixed icon background markup and z-index shenanigan.
7. Added `aria-hidden` to icons.
8. Removed hardcoded icon size in copy to clipboard component. Now icon size matches text size.
9. Changed how `spacing` is handled in menu and menu-item twig, css, and js.
10. Removed `last` logic in menu-item because it's no longer needed.
11. Updated test snaps for a bunch of components which had icon markup in their tests.

## How to test

Run the branch locally and check all the docs for Share, Menu, Copy to clipboard, and Icon. Watch out for any typos or broken demos.

### CODE REVIEW

**Comment by colbytcook** [comment]
I am seeing some extra left/right spacing on the "Icon Adding Custom Icons" section that is coming from a "o-bolt-wrapper" on the page.

File: docs-site/src/pages/pattern-lab/_patterns/40-components/share/20-share-opacity-variations.twig [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{% set schema = bolt.data.components['@bolt-components-share'].schema %}
```
Comment (colbytcook) [inline docs-site/src/pages/pattern-lab/_patterns/40-components/share/20-share-opacity-variations.twig:1]: Do we have any accessibility concerns regrading the examples here? They seem very light and might be fail contrast tests. Should we allow for these options if they do fail?

File: packages/components/bolt-share/share.schema.js [hunk 2]
```diff
@@ -59,17 +59,23 @@ module.exports = {
         },
```
Comment (colbytcook) [inline packages/components/bolt-share/share.schema.js:59]: Should the share component's **sources** prop be required?

File: docs-site/src/pages/pattern-lab/_patterns/40-components/share/20-share-opacity-variations.twig (hunk 1 above)
Comment (mikemai2awesome) [reply docs-site/src/pages/pattern-lab/_patterns/40-components/share/20-share-opacity-variations.twig:1]: They have been in use for a while now. It's up to stakeholders if they want to remove them. @colbytcook

File: packages/components/bolt-share/share.schema.js (hunk 2 above)
Comment (mikemai2awesome) [reply packages/components/bolt-share/share.schema.js:59]: I think I tried before and require doesn't work well with nested props. @colbytcook

**Comment by mikemai2awesome** [comment]
@colbytcook The icon doc has been like that. I'll address that separately when I revamp those docs.

File: docs-site/src/pages/pattern-lab/_patterns/40-components/share/20-share-opacity-variations.twig (hunk 1 above)
Comment (colbytcook) [reply docs-site/src/pages/pattern-lab/_patterns/40-components/share/20-share-opacity-variations.twig:1]: Ok, sounds good

File: packages/components/bolt-share/share.schema.js (hunk 2 above)
Comment (colbytcook) [reply packages/components/bolt-share/share.schema.js:59]: OK, then lets leave this for now

**Review by colbytcook — APPROVED** [review]

**Review by danielamorse — APPROVED** [review]
Tested and approved 👍 

When there are this many files changed, let's put docs and formatting updates (Menu and icon updates for example) in a follow-up PR, just for easier review. Thanks for bringing those components up to date!

**Pull request merged**